ocaml/forkexecd/helper/vfork_helper.c

```diff
@@ -335,14 +335,44 @@ reset_signal_handlers(void)
 static void
 clear_cgroup(void)
 {
-    int fd = open("/sys/fs/cgroup/systemd/cgroup.procs", O_WRONLY|O_CLOEXEC);
-    if (fd >= 0) {
-        char string_pid[32];
-        int ignored __attribute__((unused));
-        sprintf(string_pid, "%d\n", (int) getpid());
-        ignored = write(fd, string_pid, strlen(string_pid));
+    // list of files to try, terminated by NULL
+    static const char *cgroup_files[] = {
+        "/sys/fs/cgroup/systemd/cgroup.procs",
+        "/sys/fs/cgroup/cgroup.procs",
+        NULL
+    };
+
+    char string_pid[32];
+    int last_error = 0;
+    const char *last_error_msg = NULL;
+    const char *last_fn = NULL;
+
+    snprintf(string_pid, sizeof(string_pid), "%d\n", (int) getpid());
+
+    for (const char **fn = cgroup_files; *fn != NULL; ++fn) {
+        last_fn = *fn;
+        int fd = open(*fn, O_WRONLY|O_CLOEXEC);
+        if (fd < 0) {
+            last_error = errno;
+            last_error_msg = "Error opening file %s";
```

Comment:
We could also keep the format string constant, and customize just the operation, i.e. `Error %s file %s`.
Then if we add the proper gnu_format attributes to `error` the compiler could check at build time that we called `error` with the correct number and type of parameters.